File: dsa/Array/longestCommonPrefix.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution{
    public:
        string longestCommonPrefix(vector<string>& strs){
            if (strs.empty())
                return "";
            string current = strs[0];
            int n = strs.size();
            ;
            for (int i = 0; i < n - 1; i++) {
                string output = "";
                string temp = strs[i + 1];
                int substrLen = temp.size();
                int j = 0;

                while (substrLen > j) {
                if (temp[j] == current[j]) {
                    output += temp[j];
                } else {
                    break;
                }
                j++;
                }
                current = output;
            }
            return current;
        }
};

int main(){
    Solution s;
    vector<string> strs = {};
    string prefix = s.longestCommonPrefix(strs);
    cout << prefix << endl;
    return 0;
}

File: dsa/Array/longestCommonPrefix.py
from typing import List
class Solution:
    def longestCommonPrefix(self, strs: List[str])-> str:
        if not strs:
            return ''
        current = strs[0]
        n = len(strs)

        for i in range(n-1):
            output = ''
            temp = strs[i + 1]

            substrLen = len(temp) if temp else 0

            j=0
        
            while substrLen > j and len(current) > j:
                # print('temp[j]',  temp[j])
                # print('current[j]', current[j])
                if temp[j] == current[j]:
                    output += temp[j]
                else:
                    break
                j += 1
        
            current = output
    
        return current
    
solution = Solution()
arr = []

print(solution.longestCommonPrefix(arr))